receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -66,6 +69,15 @@ func (s *scraper) scrape(_ context.Context) (pdata.Metrics, error) {
 	for _, cpuTime := range cpuTimes {
 		s.recordCPUTimeStateDataPoints(now, cpuTime)
 	}
+
+	cpuUtilizations, err := s.ucal.Calculate(now, cpuTimes)
+	if err != nil {
+		return md, scrapererror.NewPartialScrapeError(err, metricsLen)
+	}
+	for _, cpuUtilization := range cpuUtilizations {
+		s.recordCPUUtilization(now, cpuUtilization)
+	}
```

Comment:
It looks like unnecessary slice of utilizations, should we simply pass a func pointer `recordCPUUtilization` to the `ucal` so it calls into the right func?

Reply:
This way the calculator just handles the part of calculation and the scrapper it is the responsible of recording making it more decoupled and easier to test. Still, it adds this overhead so I guess that it is a trade-off.

If you think that this kind of overhead needs to be avoided as much as possible, I can work on it :)

Reply:
What do you think @bogdandrutu ?

Reply:
I do believe that unnecessary overhead is good to be removed.

Reply:
👍  I added the `recorder` to the calculator to record it as soon as it is calculated. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7130/commits/d07241e8b42358bce47d8ac9c8cfcd168536e059

Reply:
What do you think @bogdandrutu ?

Reply:
Hi @bogdandrutu  👋 I already moved the `recordCPUUtilization` functionality inside the calculator to avoid overhead. Let me know if this looks better now 🙂

Reply:
Hi @bogdandrutu , reading the [specification discussion](https://github.com/open-telemetry/opentelemetry-specification/issues/2287) about having both metrics, seems that it can make sense to have both metrics and this could be a good candidate to be disabled by default. I have made another commit with this last change.

What do you think?